src/runtime/virtcontainers/fs_share_linux.go

```diff
@@ -407,10 +407,14 @@ func (f *FilesystemShare) UnshareFile(ctx context.Context, c *Container, m *Moun
 		}
 		// Remove the empty file or directory
 		if s.Mode().IsRegular() && s.Size() == 0 {
-			os.Remove(m.HostPath)
+			if err := os.Remove(m.HostPath); err != nil {
+				return errors.Wrapf(err, "Could not remove file from host %v", m.HostPath)
```

Comment:
Returning errors from a clean-up path is tricky. I am a relative beginner when it comes to the Kata runtime code, but the authors' intention may have been to continue the rollbackFailingContainerCreation() clean-up after such error. Logging these errors (if practical) and continuing the clean-up seems reasonable to me.

Reply:
If the intention is to ignore the error because something is not properly implemented and errors occur frequently then it is not clear to me in this part.

There is already an error returned in this function if no such HostPath exist (line 406). The error is propagated in the [call to this function](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/container.go#L550), so I don't understand why some errors are ignored and some are propagated. To me it just looks like this error check was forgotten.

Reply:
Adding one more thought - the UnshareFile function (and whatever other function) should honestly signal that something bad happened. It should be the function caller's responsibility to handle/propagate/ignore the error.